### ISSUE TITLE
MobileBertForSequenceClassification outputs super-high logits

### ISSUE THREAD

**Issue by NadiaRom** [huggingface/transformers#8938]
## Environment info

     
- `transformers` version: 4.0.0
- Platform: Ubuntu 20.04
- Python version: 3.8.5
- PyTorch version (GPU?): 1.7.0
- Tensorflow version (GPU?): None
- Using GPU in script?: True
- Using distributed or parallel set-up in script?: False

### Who can help

@LysandreJik 

## Information

Model I am using (Bert, XLNet ...): MobileBertForSequenceClassification

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [x] my own modified scripts: please see an example below

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: simple text classification using documents from sec.gov

## To reproduce

I am training a whole-text classifier with MobileBert using MobileBertForSequenceClassification:
```{python}
from transformers import MobileBertForSequenceClassification, \
                                        MobileBertTokenizerFast

ARCH = 'google/mobilebert-uncased'

model = MobileBertForSequenceClassification.from_pretrained(ARCH).cuda()
tokenizer = MobileBertTokenizerFast.from_pretrained(ARCH)

x = tokenizer(['def hello(): return "world"', 'This is some test'],
              max_length=512,
              truncation=True,
              return_tensors='pt',
              padding='longest')

with torch.no_grad():
    l = model(**x.to(model.device)).logits 
```
Resulting model outputs are extremely high:
```
tensor([[ 3289181.7500, -2371234.0000],
        [ 3198336.7500, -1882639.8750]])
```
Loading model and tokenizer with Auto- classes gives the same result. 
When using pooled output from ModileBertModel with a custom linear head (BatchNorm1d+Dropout+Linear) everything works fine.

## Expected behavior

I expect logits to be near [-3, 3], but not in 6-7 digits.

**Comment by hfawaz** [comment]
I see the same behavior when trying it out with IMDB classification. 
I solved it by passing `classifier_activation=True` for the `from_pretrained` function. 
[Documentation](https://huggingface.co/transformers/model_doc/mobilebert.html#mobilebertconfig) says it is `True` by default, however it does not seem like it.

**Comment by NadiaRom** [comment]
@hfawaz Thank you for solving this issue, it's very helpful

**Comment by jonnyzhang02** [comment]
> I see the same behavior when trying it out with IMDB classification. I solved it by passing `classifier_activation=True` for the `from_pretrained` function. [Documentation](https://huggingface.co/transformers/model_doc/mobilebert.html#mobilebertconfig) says it is `True` by default, however it does not seem like it. [EDIT] Apparently this changes the behavior of the pooling layer

I encountered the same issue and solved it by passing classifier_activation=True for the from_pretrained function. This should even be a bug in the transformers library. It is now 2025, and the result is still the same.
